staff/marti-malek/spa/scripts/view.js

```diff
@@ -0,0 +1,294 @@
+    this.__$nameInput__.val('');
+    this.__$surnameInput__.val('');
+    this.__$emailInput__.val('');
+    this.__$passwordInput__.val('');
+    this.__$passwordConfirmationInput__.val('');
+    this.__errorPanel__.message = '';
+    this.__errorPanel__.hide();
+};
+
+Object.defineProperty(RegisterPanel.prototype, 'onGoToLogin', {
+    set: function (callback) {
+        this.__$loginLink__.on('click', callback);
+    }
+});
+
+//#endregion
+
+//#region home panel
+
+function HomePanel() {
+    Panel.call(this, $('<section class="home"><h2>Welcome, <span></span>!</h2><button>Logout</button><section><section class="error"></section><ul></ul></section></section>'));
+
+    var $container = this.$element;
+
+    var $title = $container.children('h2');
+
+    var $userSpan = $title.children('span');
+    this.__$userSpan__ = $userSpan;
+
+    var $logoutButton = $container.children('button');
+    $container.append($logoutButton);
+    this.__$logoutButton__ = $logoutButton;
+}
+
+HomePanel.prototype = Object.create(Panel.prototype);
+HomePanel.prototype.constructor = HomePanel;
+
+Object.defineProperty(HomePanel.prototype, 'user', {
+    set: function (user) {
+        this.__$userSpan__.text(user.name);
+    }
+});
+
+Object.defineProperty(HomePanel.prototype, 'onLogout', {
+    set: function (callback) {
+        this.__$logoutButton__.on('click', callback);
+    }
+});
+
+//#endregion
+
+//#region search panel
+
+function SearchPanel() {
+    Panel.call(this, $('<section><form><input type="text" name="query" placeholder="..."><button type="submit">Search</button></form><ul></ul></section>'));
+
+    var $container = this.$element;
+
+    var $form = $container.children('form');
+    $container.append($form);
+    this.__$form__ = $form;
+
+    var $queryInput = $form.children('input');
+    $form.append($queryInput);
+    this.__$queryInput__ = $queryInput;
+
+    var errorPanel = new ErrorPanel;
+    $container.append(errorPanel.$element);
+    this.__errorPanel__ = errorPanel;
+
+    var $resultList = $container.children('ul');
+    $container.append($resultList);
+    this.__$resultList__ = $resultList;
+}
+
+SearchPanel.prototype = Object.create(Panel.prototype);
+SearchPanel.prototype.constructor = SearchPanel;
+
+Object.defineProperty(SearchPanel.prototype, 'onSearch', {
+    set: function (callback) {
+        this.__$form__.on('submit', function (event) {
+            event.preventDefault();
+            var query = this.__$queryInput__.val();
+
+            callback(query);
+        }.bind(this));
+    }
+});
+
+Object.defineProperty(SearchPanel.prototype, 'error', {
+    set: function (message) {
+        this.__errorPanel__.message = message;
+        this.__errorPanel__.show();
+    }
+});
+
+Object.defineProperty(SearchPanel.prototype, 'results', {
+    set: function (results) {
+        this.__$resultList__.html('');
+        this.__errorPanel__.hide();
+
+        results.forEach(function (result) {
```

Comment:
You are using JavaScript here instead of jQuery! Fix it right now!

Reply:
__errorPanel__ should not be a jquery object, I've fixed the code underneath where I did use vanilla instead of jquery though, I hadn't realize it!